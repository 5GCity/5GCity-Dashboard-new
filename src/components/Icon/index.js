/**
 * Icon Component
 * Please write a description
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import styled from 'styled-components'

export default ({ children, icon, ...props }) => (
  <Wrapper {...props}>{icon}</Wrapper>
)

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 6px;
  background-color: #89979F;

${({small})=> small ? `
  height: 32px;
  width: 32px;`
  :`
  height: 44px;
  width: 44px;`
}
`

