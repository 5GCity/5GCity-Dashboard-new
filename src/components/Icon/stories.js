/**
 * Icon Component Stories
 * Please write a description or remove this line
 *
 * @author Guilherme Patriarca <gpatriarca@ubiwhere.com>
 */
import React from 'react'
import { storiesOf } from '@storybook/react'
import { Theme } from 'globalStyles';

import Icon from './index'

const bStyle = {
  padding:'30px',
  backgroundColor: Theme.bodyBackground,
  height:'100%'
};


const exampleOne = () => (
  <div style={bStyle}>
    <Icon icon={'test'}/>
  </div>
)

const exampleTwo = () => (
  <div style={bStyle}>
    <Icon small icon={'edit'}/>
  </div>
)

storiesOf('Icon', module)
  .add('default', exampleOne)
  .add('small',exampleTwo)