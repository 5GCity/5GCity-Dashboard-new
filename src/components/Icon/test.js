/**
 * Icon Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import Icon from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<Icon />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
