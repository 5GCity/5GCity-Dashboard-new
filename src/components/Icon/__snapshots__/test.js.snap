// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 6px;
  background-color: #89979F;
  height: 44px;
  width: 44px;
}

<div
  className="c0"
/>
`;
